Code_Saeyeon_review_par_Cedric.py

```diff
@@ -0,0 +1,149 @@
+subj_accuracies  = pd.DataFrame(index = data['subject'], 
+                                columns= ['subject name', 'accuracy_estimation', 'accuracy_confidence','accuracy_estimation_pairs_shuffling','accuracy_confidence_pairs_shuffling', 
+                                          'accuracy_estimation_full_shuffling','accuracy_confidence_full_shuffling'])
+subj_accuracies['subject name'] = data['subject'].values
+
+for i in range(len(data)):
+    
+    #estimation accuracy
+    estimation_OI = data.iloc[i, data.columns.get_loc('io_pred')]#OI
+    estimation_subject = data.iloc[i, data.columns.get_loc('sub_pred')]#subject
+   
+    subj_accuracies.iloc[i, subj_accuracies.columns.get_loc('accuracy_estimation')] = math.dist((estimation_OI, estimation_subject) ,(estimation_OI, estimation_OI))
+    
+    
+    #confidence accuracy
+    confidence_OI = data.iloc[i, data.columns.get_loc('io_conf')]#OI
+    confidence_subject = data.iloc[i, data.columns.get_loc('sub_conf')]#subject
+    a = subj_correlations.loc[subj_correlations['subject name'] == data.iloc[i, 0]]['sub_io_conf_slope'].values[0]
+    b = subj_correlations.loc[subj_correlations['subject name'] == data.iloc[i, 0]]['sub_io_conf_int'].values[0]
+    confidence_predicted = confidence_OI*a + b
+    
+    subj_accuracies.iloc[i, subj_accuracies.columns.get_loc('accuracy_confidence')] = math.dist((confidence_OI, confidence_subject) ,(confidence_OI, confidence_predicted)) 
+
+      
+#compute correlations btw acc_estimation and acc_confidence
+subj_correlations_accuracies = pd.DataFrame(index=list_subject,
+                                 columns=['subject name', 'observed data', 'shuffling keeping pairs',
+                                          'full shuffling'])
+subj_correlations_accuracies['subject name'] = list_subject
+
+for subject in list_subject :
+    subj_correlations_accuracies.loc[subject]['observed data'] = np.corrcoef(subj_accuracies.loc[subj_accuracies['subject name'] == subject]['accuracy_estimation'].astype(float),
+                        subj_accuracies.loc[subj_accuracies['subject name'] == subject]['accuracy_confidence'].astype(float))[0, 1]
+
+    
+            
+'''shuffling keeping pairs and full shuffling  '''          
+
+for subject in list_subject :
+    subj_pred = data.loc[data['subject'] == subject]['sub_pred'].array #subj_pred is an array containing all the probabilty estimates of the given subject derived from the observed data      
+    io_pred = data.loc[data['subject'] == subject]['io_pred'].array
+    io_conf = data.loc[data['subject'] == subject]['io_conf'].array
+    subj_conf = data.loc[data['subject'] == subject]['sub_conf'].array
+              
+    #at each trial, compute the estimates given by the subject and by the IO (for probabilty estimates and confidence reports) 
+    array_subject = np.empty((0, 2), float) # 2D-array for the proba estimates and confidence reports of the subject
```

Comment:
ok, this is a different way than the one I suggested using permuted indices. it works too.